core/src/main/java/io/opencensus/tags/TagKey.java

```diff
@@ -14,36 +14,22 @@
 package io.opencensus.tags;
 
 import static com.google.common.base.Preconditions.checkArgument;
-import static io.opencensus.tags.TagKey.TagType.TAG_BOOLEAN;
-import static io.opencensus.tags.TagKey.TagType.TAG_LONG;
-import static io.opencensus.tags.TagKey.TagType.TAG_STRING;
 
 import com.google.auto.value.AutoValue;
+import io.opencensus.common.Function;
 import io.opencensus.internal.StringUtil;
 import javax.annotation.concurrent.Immutable;
 
-/**
- * A key to a value stored in a {@link TagContext}.
- *
- * @param <TagValueT> The type of value that can be paired with this {@code TagKey}. {@code TagKey}s
- *     can only be instantiated with types {@code String}, {@code Long}, and {@code Boolean}.
- */
+/** A key to a value stored in a {@link TagContext}. */
 @Immutable
-@AutoValue
-public abstract class TagKey<TagValueT> {
+public abstract class TagKey {
   /** The maximum length for a tag key name. */
   public static final int MAX_LENGTH = StringUtil.MAX_LENGTH;
 
-  enum TagType {
-    TAG_STRING,
-    TAG_LONG,
-    TAG_BOOLEAN
-  }
-
   TagKey() {}
 
   /**
-   * Constructs a {@code TagKey<String>} with the given name.
+   * Constructs a {@code TagKeyString} with the given name.
    *
    * <p>The name must meet the following requirements:
    *
```

Comment:
Why do you have an extra line here? Do you use https://github.com/google/google-java-format, it does a great job formatting comments.